SymphonyApi/ArmoniK.DevelopmentKit.SymphonyApi.Client/api/ArmonikSymphonyClient.cs

```diff
@@ -123,21 +125,26 @@ private void ControlPlaneConnection()
       string clientKeyFilename  = null;
       var    sslValidation      = true;
 
-      if (controlPlanSection_!.GetSection(SectionClientCertFile).Exists())
-        clientCertFilename = controlPlanSection_[SectionClientCertFile];
+      if (controlPlanSection_!.GetSection(SectionMTLS).Exists() && controlPlanSection_[SectionMTLS].ToLower() == "true")
+      {
+        if (controlPlanSection_!.GetSection(SectionClientCertFile).Exists())
+          clientCertFilename = controlPlanSection_[SectionClientCertFile];
 
-      if (controlPlanSection_!.GetSection(SectionClientKeyFile).Exists())
-        clientKeyFilename = controlPlanSection_[SectionClientKeyFile];
+        if (controlPlanSection_!.GetSection(SectionClientKeyFile).Exists())
+          clientKeyFilename = controlPlanSection_[SectionClientKeyFile];
+      }
 
       if (controlPlanSection_!.GetSection(SectionSSlValidation).Exists() &&
           controlPlanSection_![SectionSSlValidation] == "disable")
-        sslValidation = false;
+          sslValidation = false;
```

Comment:
The code above seems very similar in its goal than the one for datasynapse but done in a different way. Perhaps it will be interesting to mutualize it?